### PR TITLE
feat: extend appsettings

### PR DESCRIPTION
**Pull request by peterdeme**
- Extend app settings with new settings.
- Make `bool` -> `bool?`. If we use`NullValueHandling = NullValueHandling.Ignore`, then it needs to be a reference type, otherwise it won't be null.
Similarly to Go `Boolean` vs `*Boolean`. The first is getting serialized into `false`, the second is `null`.
- Make base class for common properties

### CODE REVIEW

File: src/stream-chat-net/AppSettings.cs [hunk 1]
```diff
@@ -1,29 +1,122 @@
...
+        public UniqueUsernameEnforcementPolicy? EnforceUniqueUsernames { get; set; }
+    }
+
+    public class AppSettings : AppSettingsBase
+    {
 
         [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "apn_config")]
         public APNConfig APNConfig { get; set; }
 
         [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "firebase_config")]
```
Comment (ferhatelmas) [inline src/stream-chat-net/AppSettings.cs:112]: missing huawei config

File: src/stream-chat-net/AppSettings.cs [hunk 2]
```diff
@@ -1,29 +1,125 @@
...
         public FirebaseConfig FirebaseConfig { get; set; }
 
-        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "webhook_url")]
-        public string WebhookURL { get; set; }
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "huawei_config")]
+        public HuaweiFields HuaweiConfig { get; set; }
```
Comment (ferhatelmas) [inline src/stream-chat-net/AppSettings.cs:116]: himm there is a difference between Fields and Config 

Fields is from GetApp endpoint to read existing configuration (it only says Enabled or not).

Config is to configure so it takes `id` and `secret`.

All push providers can take `disabled` field too to clear configs.

Comment (peterdeme) [reply src/stream-chat-net/AppSettings.cs]: Uhh 🤦 good catch. Fixed it.

**Review by ferhatelmas — APPROVED** [review]
LGTM

**Pull request merged**